src/main/java/terminator/command/CommandParser.java

```diff
@@ -0,0 +1,80 @@
+package terminator.command;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class to handle and parse user input from standard input.
+ */
+public class CommandParser {
+
+    // Any ASCII character any number of times
+    private static final String ANY_WORDS = "\\p{ASCII}*";
+
+    // dd/MM/YYYY HHmm
+    private static final String TASK_DATETIME_FORMAT = "\\d\\d?/\\d\\d?/\\d{4} \\d{4}";
+
+    private final Pattern listPattern;
+    private final Pattern markPattern;
+    private final Pattern unmarkPattern;
+    private final Pattern todoPattern;
+    private final Pattern eventPattern;
+    private final Pattern deadlinePattern;
+    private final Pattern deletePattern;
+    private final Pattern findPattern;
+
+    /**
+     * Creates a new instance of a CommandParser.
+     */
+    public CommandParser() {
+        this.listPattern = Pattern.compile("^list(\\s+" + ANY_WORDS + ")?$");
+        this.markPattern = Pattern.compile("^mark(\\s+\\d+\\s*$)?");
+        this.unmarkPattern = Pattern.compile("^unmark(\\s+\\d+\\s*$)?");
+        this.todoPattern = Pattern.compile("^todo(\\s+"
+                + ANY_WORDS
+                + ")?");
+        this.eventPattern = Pattern.compile("^event(\\s+"
+                + ANY_WORDS
+                + " /from " + TASK_DATETIME_FORMAT
+                + " /to " + TASK_DATETIME_FORMAT
+                + "\\s*$)?");
+        this.deadlinePattern = Pattern.compile("^deadline(\\s+"
+                + ANY_WORDS
+                + " /by " + TASK_DATETIME_FORMAT
+                + "\\s*$)?");
+        this.deletePattern = Pattern.compile("^delete(\\s+\\d+\\s*$)?");
+        this.findPattern = Pattern.compile("find(\\s+" + ANY_WORDS + "$)?");
+    }
+
+    /**
+     * Parses the user input and returns an instance of a concrete implementation of the Command class,
+     * which can be executed to perform the desired operation.
+     *
+     * @param input The user input to parse
+     * @return An instance of the Command class.
+     */
+    public Command parse(String input) {
+        Matcher matcher;
+        if (input.isEmpty()) {
```

Comment:
I decided to stick with `if/else` statements because `switch` statements require a target variable that is either a byte, short, char, int (and their wrappers), or String or enum. Switch statements can't match objects, so in this case it wouldn't make sense to try to match the `Matcher` object.

I feel that regex patterns are more robust and less error-prone in general for parsing user inputs, but I agree that it has performance downsides and is slower than regular switch statements.